src/material-experimental/mdc-slider/slider.ts

```diff
@@ -7,12 +7,65 @@
  */
 
 import {
+  BooleanInput,
+  coerceBooleanProperty,
+  coerceNumberProperty,
+  NumberInput
+} from '@angular/cdk/coercion';
+import {Platform} from '@angular/cdk/platform';
+import {DOCUMENT} from '@angular/common';
+import {
+  AfterViewInit,
   ChangeDetectionStrategy,
+  ChangeDetectorRef,
   Component,
+  ContentChildren,
+  ElementRef,
+  EventEmitter,
+  Inject,
+  Input,
+  OnDestroy,
+  Output,
+  QueryList,
+  ViewChild,
+  ViewChildren,
   ViewEncapsulation,
 } from '@angular/core';
-import {MDCSliderFoundation} from '@material/slider';
+import {MDCSliderFoundation, Thumb, TickMark} from '@material/slider';
 import {SliderAdapter} from './slider-adapter';
+import {MatSliderThumb} from './slider-thumb';
+
+/** A simple change event emitted by the MatSlider component. */
+export class MatSliderChange {
+  /** The MatSlider that changed. */
+  source: MatSlider;
+
+  /** The new value of the slider thumb that changed. */
+  value: number;
+
+  /** The thumb that changed. */
+  thumb: Thumb;
+
+  constructor(source: MatSlider, value: number, thumb: Thumb) {
+    this.source = source;
+    this.value = value;
+    this.thumb = thumb;
+  }
+}
+
+/** A simple interaction event emitted by the MatSlider component. */
+export class MatSliderInteraction {
```

Comment:
I plan to use it for `dragStart`, `dragEnd`, `mouseenter`, and `mouseleave` events. Since it always involves interacting with the slider thumb, we could call it `MatSliderThumbInteractionEvent`

Reply:
Side note here, should a value be part of this event? It makes sense for `dragStart` and `dragEnd` but not really for `mouseenter` and `mouseleave`.